sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CreateMultipleVirtualMachinesAndBatchQueryStatus.java

```diff
@@ -0,0 +1,138 @@
+ * - Create multiple virtual machines from PIR image with data disks
+ * - Use Azure Resource Graph to batch query the status (the number of provisioningState=Succeeded ?).
+ */
+public final class CreateMultipleVirtualMachinesAndBatchQueryStatus {
+
+    /**
+     * Main function which runs the actual sample.
+     *
+     * @param azureResourceManager instance of the azure client
+     * @param resourceGraphManager instance of the azure graph client
+     * @return true if sample runs successfully
+     */
+    public static boolean runSample(AzureResourceManager azureResourceManager, ResourceGraphManager resourceGraphManager) {
+        final Long desiredVMCount = 6L;
+        final Region region = Region.US_EAST;
+        String rgName = Utils.randomResourceName(azureResourceManager, "rg-", 15);
+        List<String> vmNameList = new ArrayList<>();
+        Long total = 0L;
+        Boolean rtnValue = false;
+        try {
+            System.out.println("Creating Resource Group: " + rgName);
+            azureResourceManager.resourceGroups().define(rgName).withRegion(region).create();
+            System.out.println("Batch creating Virtual Machines");
+            for (int i = 0; i < desiredVMCount; i++) {
+                String vmName = Utils.randomResourceName(azureResourceManager, "javascvm", 15);
+                vmNameList.add("'".concat(vmName).concat("'"));
+                azureResourceManager.virtualMachines()
+                    .define(vmName)
+                    .withRegion(region)
+                    .withExistingResourceGroup(rgName)
+                    .withNewPrimaryNetwork("10.0." + i + ".0/28")
+                    .withPrimaryPrivateIPAddressDynamic()
+                    .withoutPrimaryPublicIPAddress()
+                    .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
+                    .withRootUsername(Utils.randomResourceName(azureResourceManager, "tirekicker", 15))
+                    .withSsh(Utils.sshPublicKey())
+                    .withNewDataDisk(50, 1, CachingTypes.READ_WRITE)
+                    .withSize(VirtualMachineSizeTypes.fromString("Standard_D2a_v4"))
+                    .beginCreate();
+            }
+
+            System.out.println("Use Azure Resource Graph to batch query the status(the number of provisioningState=Succeeded).");
+            StringBuilder queryBuilder = new StringBuilder();
+            queryBuilder.append("resources ")
+                .append("| where (resourceGroup =~ ('").append(rgName).append("')) ")
+                .append("| where (type =~ ('microsoft.compute/virtualmachines')) ")
+                .append("| where (name in~ (").append(String.join(",", vmNameList)).append(")) ")
+                .append("| where (properties['provisioningState'] =~ ('Succeeded')) ")
+                .append("| project name,id,type,location,subscriptionId,resourceGroup,tags");
+            while (total < desiredVMCount) {
```

Comment:
In case there are failed VMs (provisionState = Failed) which will cause potential infinite loop, maybe we could add a query for failed instances in the `while` loop and quits if found any?

Reply:
Agree.

May remove the condition of `| where (properties['provisioningState'] =~ ('Succeeded')) `, and calcuate how many success, how many in progress, how many failed, print the number out. Break when find failed.

Reply:
Fixed in the new version.

Reply:
Fixed in the new version.